frontend/apps/reader/modules/readerlink.lua

```diff
@@ -662,19 +678,46 @@ function ReaderLink:onTap(_, ges)
     end
 end
 
---- Remember current location so we can go back to it
-function ReaderLink:addCurrentLocationToStack()
+function ReaderLink:getCurrentLocation()
+    local location
     if self.ui.document.info.has_pages then
-        table.insert(self.location_stack, self.ui.paging:getBookLocation())
+        location = self.ui.paging:getBookLocation()
     else
-        table.insert(self.location_stack, {
-            xpointer = self.ui.rolling:getBookLocation(),
-        })
+        location = {xpointer = self.ui.rolling:getBookLocation(),}
     end
+    return location
+end
+
+-- Returns true, current_location if the current location is the same as the
+-- saved_location on the top of the stack.
+-- Otherwise returns false, current_location
+function ReaderLink:compareLocationToCurrent(saved_location)
+    local current_location = self:getCurrentLocation()
+    if
+        self.ui.rolling and saved_location.xpointer
+        and saved_location.xpointer == current_location.xpointer
+    then
+        return true, current_location
+    end
+    if
+        self.ui.paging and saved_location[1] and current_location[1]
+        and current_location[1].page == saved_location[1].page
+    then
+        return true, current_location
+    end
```

Comment:
Might be @Frenzie 's preference (https://github.com/koreader/koreader/pull/10228#discussion_r1148527158), but that's even more odd than my previous remark :) We don't have anything like that anywhere, those ugly standalone `if`...
Make them one single line, there are not that long.